index.ts

```diff
@@ -72,12 +75,22 @@ function handleSerialization(input: string, api_call: (e: u32) => u32): string {
   }
 }
 
-export function debug(message: string): void {
-  handleSerialization(message, (e: u32): u32 => env.hc_debug(e));
+export function debug<T>(message: T): void {
+  if(isString<T>(message)) {
```

Comment:
I like these case statements, eases the API on the user side

Reply:
Even better is that the isString<>() functions are resolved at compile time so the build actually removes the if statement and automatically generates a unique function for each type

Reply:
Oh nice I didn't know that `instanceof` was supported. Thanks